### PR TITLE
fix: Remove next from peer dependencies; Update and fix lots of stuff.

### PR DESCRIPTION
**Pull request by yeldiRium**
This PR:

- updates many many dependencies, including major version updates
- fixes issues with new eslint rules
- adjusts next.js configs to new next.js version and webpack 5

**Please take a hard look, a lot of stuff has changed and I know very little about this package.**

### CODE REVIEW

File: lib/cli/verifyLinks/getAbsolutePath.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import untildify from 'untildify';
+// eslint-disable-next-line unicorn/import-style
```
Comment (goloroden) [inline lib/cli/verifyLinks/getAbsolutePath.ts:2]: Why is this?

File: lib/createRedirect.tsx [hunk 2]
```diff
@@ -25,6 +25,7 @@ const createRedirect = (destination: string): NextPage<RedirectProps> => {
 
   // eslint-disable-next-line @typescript-eslint/unbound-method
   RedirectComponent.getInitialProps = function ({ res }): RedirectProps {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (goloroden) [inline lib/createRedirect.tsx:28]: Why is this needed?

File: lib/services/ExecutionEnvironmentService.ts [hunk 3]
```diff
@@ -5,6 +5,7 @@ class ExecutionEnvironmentService {
     this.canUseDom = Boolean(
       /* eslint-disable @typescript-eslint/unbound-method */
       typeof window !== 'undefined' &&
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (goloroden) [inline lib/services/ExecutionEnvironmentService.ts:8]: And once again 😉

File: lib/utils/getLanguageFromUrl.ts [hunk 4]
```diff
@@ -1,9 +1,9 @@
 import { Language } from '../types/Language';
 
 const getLanguageFromUrl = function (url: string): Language {
-  const language = url.split('/').filter((item): boolean => item !== '')[0];
+  const language = url.split('/').find((item): boolean => item !== '');
```
Comment (goloroden) [inline lib/utils/getLanguageFromUrl.ts:4]: Much better 👍

I actually wonder why we did it with `filter` in the first place.

**Review by goloroden — CHANGES_REQUESTED** [review]
@yeldiRium Overall this looks pretty good 👍

I just have a few questions regarding ESLint comments.

File: lib/cli/verifyLinks/getAbsolutePath.ts (hunk 1 above)
Comment (yeldiRium) [reply lib/cli/verifyLinks/getAbsolutePath.ts:2]: Because the name `path` is already in use as a parameter and would shadow the import.

File: lib/createRedirect.tsx (hunk 2 above)
Comment (yeldiRium) [reply lib/createRedirect.tsx:28]: Because TypeScript assumes that the function `writeHead` always exists on `res`, if `res` is not undefined. When I removed the condition, the tests failed, because `writeHead` did not exist... :facepalm:

File: lib/services/ExecutionEnvironmentService.ts (hunk 3 above)
Comment (yeldiRium) [reply lib/services/ExecutionEnvironmentService.ts:8]: Again because TypeScript believes something that is not necessarily true. I could have a `window` in scope and according to the dom types it always has a `document` property, but those types might not accurately reflect the environment in which the code is being ran.

File: lib/utils/getLanguageFromUrl.ts (hunk 4 above)
Comment (yeldiRium) [reply lib/utils/getLanguageFromUrl.ts:4]: :D

**Review by goloroden — APPROVED** [review]
